x/mongo/driver/topology/server_options_test.go

```diff
@@ -0,0 +1,49 @@
+package topology
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestServerOptions(t *testing.T) {
+	t.Run("newServerConfig", func(t *testing.T) {
+		t.Parallel()
+
+		t.Run("minPoolSize should not exceed maxPoolSize", func(t *testing.T) {
+			t.Parallel()
+
+			sOpts := []ServerOption{
+				WithMaxConnections(func(uint64) uint64 {
+					return uint64(10)
+				}),
+				WithMinConnections(func(uint64) uint64 {
+					return uint64(100)
+				}),
+			}
+
+			defer func() {
+				if r := recover(); r == nil {
```

Comment:
May have to refactor these tests if we remove the `panic` and error within `ApplyURI`.